### DISCUSSION TITLE
[@types/react-router-dom] Missing types for react-router-dom and react-router v6

### DISCUSSION THREAD

**Discussion by joaocasarin** [DefinitelyTyped/DefinitelyTyped#57704] [Issues with a @types package] [answered]
Well, these libraries have recently been updated to its v6, which changes some methods, attributes. The latest version of @types for these libraries has been released 25 days ago, but it is not yet supporting the new v6.

For those trying to use latest `react-router-dom` which version is 6.1.1 and latest `@types/react-router-dom` which version is 5.3.2, you will get a lot of errors referencing the old v5 methods, which are not available on v6 anymore, such as:

* match
* PromptProps
* Prompt
* Switch
* useHistory

And many others. There are literally 38 errors like that, some may be referencing the same method/attribute.

Full list of errors attached
[log.txt](https://github.com/DefinitelyTyped/DefinitelyTyped/files/7707433/log.txt)

[a comment by an automated account is not shown]

**Comment by hamzashezad** [comment, marked as answer]
`react-router-dom` v6 is written in TypeScript and has build-in types so you don't need a DT package for it.
More here https://github.com/remix-run/react-router/releases/tag/v6.0.0-alpha.4

**Reply by joaocasarin** [reply]
Oh I haven't noticed that, thank you!
